lib/3scale/backend/configuration.rb

```diff
@@ -60,6 +72,16 @@ def configure
         ENV['CONFIG_FILE']
       ].compact)
 
+      # Assign default values to some configuration values
+      # that might been set in the config file but their
+      # environment variables not.
+      if config.master.metrics.transactions.to_s.empty?
+        config.master.metrics.transactions = CONFIG_MASTER_METRICS_TRANSACTIONS_DEFAULT
+      end
```

Comment:
In the else case you still want to make sure that the contents of `config.master.metrics.transactions` is a `String` - either call `to_s` and re-assign it before or do it in the else branch. Same thing for the setting below.

Reply:
I understand this would be needed due to currently the user can configure a Ruby datatype directly so we try to convert it to String just in case has introduced a different data type.
I have updated the code to do this.